static/js/components/ObservationPlanRequestLists.jsx

```diff
@@ -78,11 +81,89 @@ const getMuiTheme = (theme) =>
     },
   });
 
-const ObservationPlanRequestLists = ({ observationplanRequests }) => {
+const ObservationPlanGlobe = ({ observationplanRequest, loc }) => {
+  const dispatch = useDispatch();
+
+  const displayOptions = [
+    "localization",
+    "sources",
+    "galaxies",
+    "instrument",
+    "observations",
+  ];
+  const displayOptionsDefault = Object.fromEntries(
+    displayOptions.map((x) => [x, false])
+  );
+  displayOptionsDefault.localization = true;
+  displayOptionsDefault.observations = true;
+
+  const [obsList, setObsList] = useState(null);
+  useEffect(() => {
+    const fetchObsList = async () => {
+      const response = await dispatch(
+        GET(
+          `/api/observation_plan/${observationplanRequest.id}/geojson`,
+          "skyportal/FETCH_OBSERVATION_PLAN_GEOJSON"
+        )
+      );
```

Comment:
Whats the advantage of using this instead of a fetch defined in the js file?

Reply:
That assigns a state variable no?

Reply:
Yes I think you're right.